buildSrc/subprojects/binary-compatibility/src/main/groovy/gradlebuild.binary-compatibility.gradle

```diff
@@ -85,54 +82,56 @@ dependencies {
         to.attribute(ARTIFACT_TYPE, 'gradle-classpath')
     }
 
-    // TODO update this to use a dependency on :distributionFull instead of 'rootProject.subprojects { }'
     rootProject.subprojects { subproject ->
-        if (subproject.name == 'distributionsDependencies') {
-            currentClasspath platform(subproject)
-        }
         plugins.withId('gradlebuild.distribution.api') {
             apiSourceFolders.from(subproject.sourceSets.main.allJava.sourceDirectories)
             if (subproject.plugins.hasPlugin("kotlin")) {
                 apiSourceFolders.from(subproject.sourceSets.main.kotlin.sourceDirectories)
             }
-            newArchivesFiles.from(subproject.jar)
-            // TODO: Current classpath should be derived from the other "Gradle runtime" sources like api-metadata, runtime-api, distribution building, etc...
-            currentClasspath project(subproject.path)
-
-            // This transform uses the result of the exploded zip extraction
-            // and returns a single jar file based on the lookup project name
-            String projectName = subproject.moduleIdentity.baseName.get() - 'gradle-'
-            registerTransform(FindGradleJar) {
-                from.attribute(ARTIFACT_TYPE, 'gradle-libs-dir')
-                to.attribute(ARTIFACT_TYPE, projectName)
-                parameters {
-                    target = projectName
-                }
-            }
-            def appendixName = projectName.split('-')*.capitalize().join('')
-            def baselineConf = configurationsContainer.create("japicmp-baseline-${appendixName}") {
-                extendsFrom configurationsContainer.baseline
-                attributes.attribute(ARTIFACT_TYPE, projectName)
-            }
-            oldArchivesFiles.from(baselineConf)
         }
     }
 }
 
 def checkBinaryCompatibility = tasks.register("checkBinaryCompatibility", JapicmpTask) {
+    def oldArchivesFiles = objects.fileCollection()
+    def newArchivesFiles = objects.fileCollection()
     def baseVersion = moduleIdentity.version.map { it.baseVersion.version }
     def isSnapshot = moduleIdentity.snapshot
+    def currentClasspath = configurations.currentClasspath.resolvedConfiguration.lenientConfiguration
```

Comment:
Interesting. I have always used `incoming.artifactView { lenient(true) }.files ` for this.

E.g.:
https://github.com/gradle/gradle/blob/fc5a10e07409595e1020cf7948b9d05646e6371a/buildSrc/subprojects/packaging/src/main/kotlin/gradlebuild/shaded-jar.gradle.kts#L182

"outgoing" and "incoming" are the _newer_ APIs. But I think under the hood it'll be the same result. You might change it here for consistency, but we can also keep it as you propose.